relayer/chain/parachain/header_proof.go

```diff
@@ -0,0 +1,73 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/snowfork/go-substrate-rpc-client/v3/types"
+	"github.com/wealdtech/go-merkletree"
+)
+
+func CreateParachainHeaderProof(allParaHeads []types.Bytes, ourParaHead types.Header, expectedRoot types.H256) ([][32]byte, error) {
+	ourParaHeadBytes, err := types.EncodeToBytes(ourParaHead)
+	if err != nil {
+		return [][32]byte{}, err
+	}
+
+	paraTreeData := make([][]byte, len(allParaHeads))
+	for i, paraHead := range allParaHeads {
+		paraTreeData[i] = paraHead
+	}
+
+	// Create the tree
+	paraMerkleTree, err := merkletree.NewUsing(paraTreeData, &Keccak256{}, nil)
+	if err != nil {
+		return [][32]byte{}, err
+	}
+
+	// Generate Merkle Proof for our parachain's head
+	proof, err := paraMerkleTree.GenerateProof(ourParaHeadBytes)
```

Comment:
The library used here may create a different merkle tree from our solidity implementation, see comment above. Let’s see what Thomasz implements and then adjust.